cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaBrokerConfigurationBuilderTest.java

```diff
@@ -1412,6 +1412,121 @@ public void testOAuthOptions()  {
+                .withSasl(true)
+                .withListenerConfig(Map.of())
+                .endKafkaListenerAuthenticationCustomAuth()
+                .build();
+
+        String configuration = new KafkaBrokerConfigurationBuilder()
+                .withListeners("my-cluster", "my-namespace", singletonList(listener), false)
+                .build();
+
+        assertThat(configuration, configuration.contains("listener.security.protocol.map=CONTROLPLANE-9090:SSL,REPLICATION-9091:SSL,CUSTOM-LISTENER-9092:SASL_SSL"));
+    }
+
+    @ParallelTest
+    public void testCustomConfigSetProtocolMapCorrectlyForPlainSasl() {
+        GenericKafkaListener listener = new GenericKafkaListenerBuilder()
+                .withName("CUSTOM-LISTENER")
+                .withPort(9092)
+                .withType(KafkaListenerType.INTERNAL)
+                .withTls(false)
+                .withNewKafkaListenerAuthenticationCustomAuth()
+                .withSasl(true)
+                .withListenerConfig(Map.of())
+                .endKafkaListenerAuthenticationCustomAuth()
+                .build();
+
+        String configuration = new KafkaBrokerConfigurationBuilder()
+                .withListeners("my-cluster", "my-namespace", singletonList(listener), false)
+                .build();
+
+        assertThat(configuration, configuration.contains("listener.security.protocol.map=CONTROLPLANE-9090:SSL,REPLICATION-9091:SSL,CUSTOM-LISTENER-9092:SASL_PLAINTEXT"));
+    }
+
+    @ParallelTest
+    public void testCustomConfigSetProtocolMapCorrectlyForPlain() {
+        GenericKafkaListener listener = new GenericKafkaListenerBuilder()
+                .withName("CUSTOM-LISTENER")
+                .withPort(9092)
+                .withType(KafkaListenerType.INTERNAL)
+                .withTls(false)
+                .withNewKafkaListenerAuthenticationCustomAuth()
+                .withSasl(false)
+                .withListenerConfig(Map.of())
+                .endKafkaListenerAuthenticationCustomAuth()
+                .build();
+
+        String configuration = new KafkaBrokerConfigurationBuilder()
+                .withListeners("my-cluster", "my-namespace", singletonList(listener), false)
+                .build();
+
+        assertThat(configuration, configuration.contains("listener.security.protocol.map=CONTROLPLANE-9090:SSL,REPLICATION-9091:SSL,CUSTOM-LISTENER-9092:PLAINTEXT"));
+    }
+
+    @ParallelTest
+    public void testCustomConfigPrefixesUserProvidedConfig() {
+        GenericKafkaListener listener = new GenericKafkaListenerBuilder()
+                .withName("CUSTOM-LISTENER")
+                .withPort(9092)
+                .withType(KafkaListenerType.INTERNAL)
+                .withTls(true)
+                .withNewKafkaListenerAuthenticationCustomAuth()
+                .withSasl(true)
+                .withListenerConfig(Map.of("oauthbearer.sasl.client.callback.handler.class", "client.class",
```

Comment:
Can we add some `ssl...` option here to test that it will not be passed into the configuration?

Reply:
In this present case, I've moved the test to `KafkaClusterTest` (https://github.com/strimzi/strimzi-kafka-operator/pull/6162/files#diff-a6f8b697177e22f1da12401caa919e111ac8d6b204b437022576893e04732784R3698-R3722) as the removal of forbidden prefixes happens in `KafkaCluster` itself, rather than `KafkaBrokerConfigurationBuilder`. That said, if we move the removal to `KafkaBrokerConfigurationBuilder` we can then test here!